arbor/backends/gpu/event_stream.hpp

```diff
@@ -84,6 +89,49 @@ class event_stream : public event_stream_base<Event, typename memory::device_vec
         arb_assert(num_events == base::ev_data_.size());
     }
 
+    friend void serialize(serializer& ser, const std::string& k, const event_stream<Event>& t) {
```

Comment:
Wouldn't it be less error-prone if the base class template `event_stream_base` had its own serialization hooks which could be called from here?

Reply:
Hmm. It's a tradeoff. In the current way I can use the `ENABLE` macro in the `multicore` subclass
and need a custom serializer in the `gpu` case. Your proposal is slightly more code to write:
1. `ENABLE` in base
2. custom one-liner in `multicore`
3. custom serializer in `gpu` minus the base members plus one line for serializing `base`

Seen that way: Yes, probably better ;)